message_hub/src/main/java/ordina/youbionic/infrastructure/RabbitMQPublisher.java

```diff
@@ -0,0 +1,49 @@
+package ordina.youbionic.infrastructure;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import ordina.youbionic.Exception.IllegalEnumValueException;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+public class RabbitMQPublisher {
+    private final Channel channel;
+
+    public RabbitMQPublisher() throws Exception{
```

Comment:
@Requiredargsconstructor can solve the constructor making.

So it looks a little bit more readable, and if the configuration is needed what you are doing in the constructor than you should think of moving this configuration to another class because this class is not responsible for the configuration of the channel or topic stuff